src/_js/map.js

```diff
@@ -200,24 +200,48 @@
 		});
 
 		const mapContainer = document.querySelector('#map');
-		const MARKERS_URL_PATH = mapContainer.dataset.markerPack ? mapContainer.dataset.markerPack : 'markers.json';
-		const xhr = new XMLHttpRequest();
-		xhr.open('GET', URL_PREFIX + MARKERS_URL_PATH);
-		xhr.responseType = 'json';
-		xhr.onload = function() {
-			if (xhr.status === 200) {
-				xhr.response.forEach(m => {
-					const options = {'title': m.description};
-					if (m.icon && m.icon in icons) { options.icon = icons[m.icon]; }
-					if (!_this.markersLayers[m.z]) { _this.markersLayers[m.z] = new L.layerGroup(); }
-					_this.markersLayers[m.z].addLayer(
-						L.marker(_this.map.unproject([m.x + 0.5, m.y + 0.5], 0), options)
-					);
-				});
-				_this._tryShowMarkers();
-			}
-		};
-		xhr.send();
+		const urlParams = new URLSearchParams(window.location.search);
+		// Possible markers sources
+		// A) https://example.com?m=<base64-json-str>#32368,32198,7:0
+		// B) https://example.com?mf=https://example.com/pack.json#32368,32198,7:0
+		// C) <div id="map" data-marker-json="<json-str>" ...>
+		// D) <div id="map" data-marker-pack="https://example.com/pack.json" ...>
+		// E) fallback: https://tibiamaps.github.io/tibia-map-data/markers.json
+		if (urlParams.get('m')) {
+			buildMarkerLayers(atob(JSON.parse(urlParams.get('m'))));
+		} else if (urlParams.get('mf')) {
+			loadMarkersPack(urlParams.get('mf'));
+		} else if (mapContainer.dataset.markerJson) {
+			buildMarkerLayers(JSON.parse(mapContainer.dataset.markerJson));
+		} else if (mapContainer.dataset.markerPack) {
+			loadMarkersPack(URL_PREFIX + mapContainer.dataset.markerPack);
+		} else {
+			loadMarkersPack(URL_PREFIX + 'markers.json');
+		}
```

Comment:
I think it would be nice to extract this logic to a separate function (called `getMarkerPackUrl` or some such) that returns a URL to load, so we could have early `return`s instead of all the `else if`s.

Reply:
I don't think this could be much cleaner given the hierarchy I proposed (A behavior overrides B, B overrides C, etc.), but I gave it a try on my latest changes. Lemme know if it looks better. We can also consider other hierarchy or removing some of the sources.